services/powchain/src/main/java/tech/pegasys/teku/services/powchain/PowchainConfiguration.java

```diff
@@ -180,6 +182,16 @@ public Builder depositSnapshotPath(final String depositSnapshotPath) {
       return this;
     }
 
+    public Builder depositSnapshotPathFromCheckpointSyncUrl(final String checkpointSyncUrl) {
+      if (StringUtils.isNotBlank(checkpointSyncUrl)) {
+        this.depositSnapshotPath =
+            Optional.of(UrlSanitizer.appendPath(checkpointSyncUrl, DEPOSIT_SNAPSHOT_URL_PATH));
+        return this;
+      }
+      throw new IllegalStateException(
```

Comment:
This should be a `InvalidConfigurationException`

Reply:
But there should be no configuration that would allow that? Since I did checks before using that depositSnapshotPathFromCheckpointSyncUrl method

Reply:
If this is the case, the reason for the exception is wrong. Do not mention the cli option, just the parameter name:

e.g. `checkpointSyncUrl cannot be null/empty`.